### PR TITLE
Raw JSON RPC getrawtransaction

### PR DESCRIPTION
**Pull request by 0x4c1d**
Partially #247 
Implemented `getrawtransaction`. Although marked what has been done, in fact it was not.
I will be glad to any feedback. thank

### CODE REVIEW

**Comment by 0x4c1d** [comment]
[clabot:check]

File: rpc/src/v1/impls/raw.rs [hunk 1]
```diff
@@ -98,6 +107,105 @@ impl RawClientCoreApi for RawClientCore {
...
+			None => return Err(transaction_not_found(hash)),
+		};
+
+		let transaction_bytes = serialize(&transaction);
+		let raw_transaction = RawTransaction::new(transaction_bytes.take());
+
+		if verbose {
+			let meta = match self.storage.transaction_meta(&hash) {
+				Some(meta) => meta,
+				None => return Err(transaction_of_side_branch(hash)),
```
Comment (NikVolf) [inline rpc/src/v1/impls/raw.rs]: isn't it vice versa, this is `Err(transaction_not_found(hash))` and `Err(transaction_of_side_branch(hash))` is below?

File: rpc/src/v1/impls/raw.rs [hunk 2]
```diff
@@ -98,6 +107,105 @@ impl RawClientCoreApi for RawClientCore {
...
+			};
+
+			let block_header = match self.storage.block_header(meta.height().into()) {
+				Some(block_header) => block_header,
+				None => return Err(transaction_not_found(hash)),
+			};
+
+			let best_block = self.storage.best_block();
+			if best_block.number < meta.height() {
+				return Err(transaction_not_found(hash));
```
Comment (NikVolf) [inline rpc/src/v1/impls/raw.rs]: this is probably side branch as well (if possible at all)

File: rpc/src/v1/impls/raw.rs (hunk 1 above)
Comment (0x4c1d) [reply rpc/src/v1/impls/raw.rs]: hmm, I did it like this [hmm, I did it like this https://github.com/parityTech/parity-bitcoin/blob/master/rpc/src/v1/impls/blockchain.rs#L122
If necessary, correct?](https://github.com/parityTech/parity-bitcoin/blob/master/rpc/src/v1/impls/blockchain.rs#L122)
If necessary, correct?

Comment (NikVolf) [reply rpc/src/v1/impls/raw.rs]: probably incorrect there as well @svyatonik ?

Comment (svyatonik) [reply rpc/src/v1/impls/raw.rs]: Imo - everything is correct - transaction meta is inserted when block is canonized => missing meta means that tx isn't canonized (i.e. from side branch). The fact that tx exists is checked couple of lines above.

**Comment by svyatonik** [comment]
@General-Beck Looks like CI issue is still here - the build takes forever to complete?

**Comment by svyatonik** [comment]
@NikVolf Could you, please, sign off the PR or request changes when you have a moment? :) Gitlab is sick (@General-Beck has promised to look at it once the PR is merged), but I ack that the branch compiles :) Thanks!

**Pull request merged**